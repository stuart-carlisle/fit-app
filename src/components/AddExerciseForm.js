import React, { useState , useEffect} from 'react'
import moment from 'moment'
import { connect } from 'react-redux'
import numeral from 'numeral'

export const AddExerciseForm = ({
  filters = {},
  onSubmit, 
  exercise = {
    description:'',
    type:'',
    energy:'',
    time:''
  }
}) => {
    const defaultEnergy = numeral(exercise.energy).value()
    const defaultTime = numeral(exercise.time).value()
    const date = filters.date
    const referenceDatabase = filters.referenceDatabase
    const [time, setTime] = useState(defaultTime)
    const [error, setError] = useState('')
    const [energy, setEnergy] = useState(defaultEnergy)
    const diaryDate = date ? date.valueOf() : moment().startOf('day').valueOf()
    const lastUsed = moment().valueOf()

    useEffect(()=>{
        
      let refreshed = JSON.parse(localStorage.getItem('refreshed'))
      if(refreshed){
        window.history.back()
        window.scrollTo(0,0)
        localStorage.removeItem('refreshed')
      }else{
        localStorage.setItem('refreshed',JSON.stringify(true))
      }
    },[])

    const onFormSubmit = (e) => {
        e.preventDefault();
        const exerciseData = { 
          description: exercise.description,
          type: exercise.type
        }
        if (numeral(time).value()<=0) {
          setError('Please provide a time greater than 0')
          window.scrollTo(0,0)
        } else {
          setError('')
          onSubmit(
            referenceDatabase,
            diaryDate,
            lastUsed,
            exerciseData,
            time.toString(),
            energy.toString()
          )
          localStorage.removeItem('refreshed')
        }
      }
    
    const onTimeChange = (e)=>{
        const time = e.target.value
          if (!time || time.match(/^\d{1,}(\.\d{0,1})?$/)) {
          setTime(time)
          const timeValue = numeral(time).value()
          setEnergy(Math.round(defaultEnergy * (timeValue/defaultTime)))
        }
    }

    const onEnergyChange = ()=>{
      
    }

    return (
        <div>
        <div className="content-container__background">
        <div className="content-container__form__update">
            <form className= "form" onSubmit={onFormSubmit}>
                {error && <p id="error-message" className="form__error">{error}</p>}
                  <h1 className="page-header__title-update"> {exercise.description} </h1>
                    <span className="text-input__span-container-form">
                      <div className="content-container__update-titles">
                        <h3 className="form__heading__update">Exercise Time (mins):</h3>
                        <h3 className="form__heading__update">Energy (calories): </h3>
                      </div>
                      <div className="content-container__update">
                        <input  //Exercise Duration
                          className="text-input__update-form"
                          type="text"
                          value={time}
                          onChange={onTimeChange}
                          placeholder="Enter a Duration"
                        />  
                        <input  //Energy Used
                          className="text-input__update-form"
                          type="text"
                          value={energy?energy:''}
                          onChange={onEnergyChange}
                          placeholder="Energy Used"
                        /> 
                      </div>
                    </span>
                <button className="button button--form">Add Exercise</button>
            </form>
          </div>
        </div>
      </div>  
    )
}

const mapStateToProps = (state) => {
  return {
    filters: state.filters
  }
}


export default connect(mapStateToProps)(AddExerciseForm);