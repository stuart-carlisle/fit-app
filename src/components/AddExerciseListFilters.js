import React, { useState } from 'react';
import { connect } from 'react-redux';
import { setTextFilter, sortByDate, sortByAlphabetical, setReferenceDatabaseToPublic } from '../actions/filters';
import { setExercisesPublic } from '../actions/exercises'
import algoliasearch from 'algoliasearch/lite'

export const AddExerciseListFilters = ({
    filters,
    setTextFilter, 
    setReferenceDatabasePublic,
    sortByDate, 
    sortByAlphabetical,
    setExercisesPublic 
  }) => {
 
  const searchClient = algoliasearch (
      process.env.ALGOLIA_APP_ID,
      process.env.ALGOLIA_API_KEY
  )

  const index = searchClient.initIndex('exercises');

  const [text, setText] = useState('')
    
  const onTextChange = (e) => {
    setTextFilter(e.target.value)
    setText(e.target.value)
  }

  const onSortChange = (e) => {
    if (e.target.value === 'date') {
      sortByDate()
    } else if (e.target.value === 'alphabetical') {
      sortByAlphabetical()
    }
  }

  const onSubmit = (e) => {
    e.preventDefault()
    index
      .search({
        query: text
      })
      .then((responses) => {
        const array = responses.hits
        const newExercises = array.map((exercise)=>{
          return{
            description:exercise.description,
            type:exercise.type,
            energy:exercise.energy,
            time:exercise.time,
            id:exercise.objectID
          }
        })
        setExercisesPublic(newExercises)
      }).then(()=>{
        setReferenceDatabasePublic()
      })
}

    return (
      
      <div className="header__content--filters"> 
      <form className="filters-text-input__span-container" onSubmit={onSubmit}>
          <div className="input-group__item-page-header">
            <input
              type="text"
              className="filters-text-input"
              placeholder="SEARCH EXERCISES"
              value={text}
              onChange={onTextChange}
            />
          </div>
          <div className="input-group__item-page-header">
            <select
              className="minimal"
              value={filters.sortBy}
              onChange={onSortChange}
            >
              <option value="date">DATE</option>
              <option value="alphabetical">ALPHABETICAL</option>
            </select>
          </div>
            <button className="button filters-button--form-page-header">Search Online</button>
            </form>   
        </div>
    )
}

const mapStateToProps = (state) => {
  return {
    filters: state.filters
  };
};

const mapDispatchToProps = (dispatch) => ({
  setTextFilter: (text) => dispatch(setTextFilter(text)),
  sortByDate: () => dispatch(sortByDate()),
  sortByAlphabetical: () => dispatch(sortByAlphabetical()),
  setReferenceDatabasePublic: () => dispatch(setReferenceDatabaseToPublic()),
  setExercisesPublic: (exercises) => dispatch(setExercisesPublic(exercises))
})

export default connect(mapStateToProps,mapDispatchToProps)(AddExerciseListFilters);

