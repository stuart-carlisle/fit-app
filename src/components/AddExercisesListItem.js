import React from 'react';
import { Link } from 'react-router-dom';

export const AddExercisesListItem = ({ description, energy, id }) => (


  <Link className="list-item" to={`/add-exercise-confirmation/${id}`}>
    <div>
      <p className="list-item__title">{description}</p>
    </div>
    <div>
      <p className="list-item__data">{energy}</p>
    </div>  
  </Link>

);

export default AddExercisesListItem;
