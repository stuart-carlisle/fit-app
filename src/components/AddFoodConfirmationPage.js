import React from 'react'
import { connect } from 'react-redux'
import AddFoodForm from './AddFoodForm'
import { Link } from 'react-router-dom'
import { startAddFoodDiary, startAddFoodPrivate, startUpdateFoodPrivate } from '../actions/foods'

export const AddFoodConfirmationPage = (props) => {
  
  const onSubmit = (referenceDatabase, diaryDate, lastUsed, numberOfServings, food, meal) => {
    if (referenceDatabase==='public'){
      props.startAddFoodPrivate({...food,lastUsed})
    } else if (referenceDatabase==='private'){
      props.startUpdateFoodPrivate(props.food.id,{...food,lastUsed})
    }
    props.startAddFoodDiary({...food, diaryDate, numberOfServings, meal})
    props.history.push('/diary');
  }
  

  return (
     <div>
     <Link className="button__back-button" to={`/add`}><img className="image__header" src="/images/back-arrow.svg"/></Link>
        <div className="page-header__filter page-header__filter--with-back">
          <div className="content-container">
            <h1 className="page-header__title">ADD FOOD</h1>
          </div>
        </div>
        <div className="show-for-mobile page-header__margin"></div>
        <div className="content-container">
        <AddFoodForm id="add-food-form" targets={props.targets} food={props.food} onSubmit={onSubmit} />
        </div>
     </div>
 )
}

const mapStateToProps = (state,props) => {
  if(state.filters.referenceDatabase==='public'){
    return {
      food: state.foodsPublic.find((food) => food.id === props.match.params.id),
      targets: state.targets
    }
  }else if(state.filters.referenceDatabase==='private'){
    return {
      food: state.foodsPrivate.find((food) => food.id === props.match.params.id),
      targets: state.targets
    }
  }
}

const mapDispatchToProps = (dispatch) =>({
    startAddFoodDiary: (food) => dispatch(startAddFoodDiary(food)),
    startUpdateFoodPrivate: (id, food) => dispatch(startUpdateFoodPrivate(id, food)),
    startAddFoodPrivate: (food) => dispatch(startAddFoodPrivate(food))
})

export default connect(mapStateToProps, mapDispatchToProps)(AddFoodConfirmationPage);




