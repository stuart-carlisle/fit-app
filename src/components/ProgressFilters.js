import React, { useState, useEffect } from 'react';
import { connect } from 'react-redux';
import { setChartType } from '../actions/filters';

export const ProgressFilters = ({
    setChartType
  }) => {

  const [chartTypeInput, setChartTypeInput] = useState('calorie-tracker')
  const [option, setOption] = useState('week')
  
  const [option1, setOption1] = useState('Last Week')
  const [option2, setOption2] = useState('Last 2 Weeks')
  const [option3, setOption3] = useState('2 Months')

  const [option1Value, setOption1Value] = useState('week')
  const [option2Value, setOption2Value] = useState('fortnight')
  const [option3Value, setOption3Value] = useState('2months')

  const onSubmit = (e) => {
    e.preventDefault()
    setChartType({
      chart: chartTypeInput,
      option: option
    })
  }

  useEffect(()=>{
    setChartType({
      chart: chartTypeInput,
      option: option
    })
  },[])

  const onChartTypeChange = (e) => {
    if(e.target.value==='calorie-tracker'){
      setOption1('Last Week')
      setOption2('Last 2 Weeks')
      setOption3('2 Months')
      setOption1Value('week')
      setOption2Value('fortnight')
      setOption3Value('2months')
      setOption('week')
    }else if(e.target.value==='weight-tracker'){
      setOption1('Last 3 Months')
      setOption2('Last 6 Months')
      setOption3('Last Year')
      setOption1Value('3months')
      setOption2Value('6months')
      setOption3Value('year')
      setOption('3months')
    }
    setChartTypeInput(e.target.value) 
  }

  const onOptionChange = (e) => {
    setOption(e.target.value)
  }

  const onBackClick = (e) => {
    e.preventDefault()
    window.history.back()
          window.scrollTo(0,0)
  }

    return (
      <div>
      <button className="button__back-button" onClick={onBackClick}><img className="image__header" src="/images/back-arrow.svg"/></button>
      <div className="page-header__filter page-header__filter--with-back">
      <div className="content-container">
      <div className="header__content--filters"> 
        <form className="filters-text-input__span-container" onSubmit={onSubmit}>
          <div className="input-group__item-page-header">
            <select
              className="minimal minimal--chart-filter"
              value={chartTypeInput}
              onChange={onChartTypeChange}
            >
              <option value="calorie-tracker">CALORIE TRACKER</option>
              <option value="weight-tracker">WEIGHT TRACKER</option>
            </select>
            </div>
          <div className="input-group__item-page-header">
          <select
            className="minimal minimal--chart-filter"
            value={option}
            onChange={onOptionChange}
          > 
            <option value={option1Value}>{option1}</option>
            <option value={option2Value}>{option2}</option>
            <option value={option3Value}>{option3}</option>
          </select>
          </div>
          <button className="button filters-button--form-page-header">Submit</button>
        </form>
        </div>
      </div>
      </div>
      </div>
    )
}

const mapDispatchToProps = (dispatch) => ({
  setChartType: (chartType) => dispatch(setChartType(chartType))
})

export default connect(undefined,mapDispatchToProps)(ProgressFilters);

// <OutsideClickHandler
//             onOutsideClick={()=>{setCalendarFocused(null)}}>
//               <DateRangePicker 
//                 startDate={filters.startDate}
//                 endDate={filters.endDate}
//                 onDatesChange={onDatesChange}
//                 focusedInput={calendarFocused}
//                 onFocusChange={onFocusChange}
//                 showClearDates={true}
//                 numberOfMonths={1}
//                 isOutsideRange={() => false}
//               />
//             </OutsideClickHandler>

//const [calendarFocused, setCalendarFocused] = useState(null)

// const onDatesChange = ({ startDate, endDate }) => {
//   setStartDate(startDate);
//   setEndDate(endDate);
// };

// const onFocusChange = ( focused ) => {
//   setCalendarFocused(focused);
//}

//import { DateRangePicker } from 'react-dates';

//import OutsideClickHandler from 'react-outside-click-handler'