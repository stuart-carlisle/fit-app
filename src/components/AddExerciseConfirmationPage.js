import React from 'react'
import { connect } from 'react-redux'
import AddExerciseForm from './AddExerciseForm'
import { Link } from 'react-router-dom'
import { startAddExerciseDiary, startAddExercisePrivate, startUpdateExercisePrivate } from '../actions/exercises'

export const AddExerciseConfirmationPage = (props) => {
  
  const onSubmit = (referenceDatabase, diaryDate, lastUsed, exerciseDescAndType, time, energy) => {
    if (referenceDatabase==='public'){
      props.startAddExercisePrivate({...exerciseDescAndType, time, energy, lastUsed})
    } else if (referenceDatabase==='private'){
      props.startUpdateExercisePrivate(props.exercise.id,{...exerciseDescAndType, time, energy, lastUsed})
    }
    props.startAddExerciseDiary({...exerciseDescAndType, time, diaryDate, energy})
    props.history.push('/diary');
  }
  

  return (
     <div>
     <Link className="button__back-button" to={`/add-exercise`}><img className="image__header" src="/images/back-arrow.svg"/></Link>
        <div className="page-header__filter page-header__filter--with-back">
          <div className="content-container">
            <h1 className="page-header__title">ADD EXERCISE</h1>
          </div>
        </div>
        <div className="show-for-mobile page-header__margin"></div>
        <div className="content-container">
        <AddExerciseForm id="add-exercise-form" exercise={props.exercise} onSubmit={onSubmit} />
        </div>
     </div>
 )
}

const mapStateToProps = (state,props) => {
  if(state.filters.referenceDatabase==='public'){
    return {
      exercise: state.exercisesPublic.find((exercise) => exercise.id === props.match.params.id)
    }
  }else if(state.filters.referenceDatabase==='private'){
    return {
      exercise: state.exercisesPrivate.find((exercise) => exercise.id === props.match.params.id)
    }
  }
}

const mapDispatchToProps = (dispatch) =>({
    startAddExerciseDiary: (exercise) => dispatch(startAddExerciseDiary(exercise)),
    startUpdateExercisePrivate: (id, exercise) => dispatch(startUpdateExercisePrivate(id, exercise)),
    startAddExercisePrivate: (exercise) => dispatch(startAddExercisePrivate(exercise))
})

export default connect(mapStateToProps, mapDispatchToProps)(AddExerciseConfirmationPage);




