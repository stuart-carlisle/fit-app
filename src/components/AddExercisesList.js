import React from 'react'
import { connect } from 'react-redux'
import AddExercisesListItem from './AddExercisesListItem'
import selectExercises from '../selectors/exercises'

export const AddExercisesList = ({exercises}) => {
  let exercisesTopResults
  
  if(exercises.length>10){
    exercisesTopResults = exercises.slice(0,10)
  }else{
    exercisesTopResults = exercises
  }

  return(<div className="content-container">
   <div className="list-header">
    <div className="">Exercise</div>
    <div className="show-for-tablet">Calories Burned</div>
    <div className="show-for-mobile">Cals Burned</div>
   </div>
    <div className="list-body">
      {
        exercisesTopResults.length === 0 ? (
          <div className="list-item list-item--message">
            <span>No Exercises In History</span>
          </div>
        ) : (
         exercisesTopResults.map((exercise) => {
            return <AddExercisesListItem key={exercise.id} {...exercise} />
          })
        )
      }
    </div>
  </div>  
  )
}

const mapStateToProps = (state) => {
  return {
    exercises: state.filters.referenceDatabase === 'public' ? state.exercisesPublic : selectExercises(state.exercisesPrivate,  state.filters)
  }
}

export default connect(mapStateToProps)(AddExercisesList);
