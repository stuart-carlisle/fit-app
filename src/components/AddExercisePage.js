import React, { useEffect } from 'react'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
import AddExerciseListFilters from './AddExerciseListFilters'
import AddExercisesList from './AddExercisesList'
import { setTextFilter, sortByDate, setReferenceDatabaseToPrivate } from '../actions/filters'

export const AddExercisePage = ({ setTextFilter, sortByDate, setReferenceDatabaseToPrivate}) => {
  useEffect(()=>{
    localStorage.removeItem('refreshed')
      setTextFilter('')
      sortByDate()
      setReferenceDatabaseToPrivate()
  },[])
    return(
      <div>
      <Link className="button__back-button" to={`/diary`}><img className="image__header" src="/images/back-arrow.svg"/></Link>
    <div className="page-header__filter page-header__filter--with-back">
      <div className="content-container">
        <AddExerciseListFilters />
      </div>  
    </div>
    <AddExercisesList />
    <div className="content-container">
    <Link className="button button--link" to={`/create-exercise`}>+ CREATE CUSTOM EXERCISE</Link>
    </div>
    </div>
    )
}

const mapDispatchToProps = (dispatch) => ({
  setTextFilter: (text) => dispatch(setTextFilter(text)),
  sortByDate: () => dispatch(sortByDate()),
  setReferenceDatabaseToPrivate: () => dispatch(setReferenceDatabaseToPrivate())
})

export default connect (undefined, mapDispatchToProps)(AddExercisePage)