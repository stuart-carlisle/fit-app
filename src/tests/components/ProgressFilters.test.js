import React from 'react'
import { shallow } from 'enzyme'
import { ProgressFilters } from '../../components/ProgressFilters'

let wrapper, setChartType

beforeEach(()=>{
    setChartType=jest.fn()
    wrapper = shallow(
        <ProgressFilters
            setChartType={setChartType}
        />
    )
})

test('should render the progress filters correctly', ()=>{
    expect(wrapper).toMatchSnapshot()
})

test('Should handle chart type change', ()=>{
    const value = 'weight-tracker'
    wrapper.find('select').at(0).simulate('change',{
        target:{
            value
        }
    })
    expect(wrapper.find('select').at(0).prop('value')).toBe(value)
})

test('Should handle chart type option change', ()=>{
    const value = '3months'
    wrapper.find('select').at(1).simulate('change',{
        target:{
            value
        }
    })
    expect(wrapper.find('select').at(1).prop('value')).toBe(value)
})

test('Should render correct options list when chart type is changed to weight-tracker',()=>{
    const value = 'weight-tracker'
    wrapper.find('select').at(0).simulate('change',{
        target:{
            value
        }
    })
    expect(wrapper.find('option').at(2).prop('value')).toBe('3months')
    expect(wrapper.find('option').at(2).text()).toBe('Last 3 Months')
    expect(wrapper.find('option').at(3).prop('value')).toBe('6months')
    expect(wrapper.find('option').at(3).text()).toBe('Last 6 Months')
    expect(wrapper.find('option').at(4).prop('value')).toBe('year')
    expect(wrapper.find('option').at(4).text()).toBe('Last Year')
})

test('Should render correct options list when chart type is changed to weight-tracker',()=>{
    let value = 'weight-tracker'
    wrapper.find('select').at(0).simulate('change',{
        target:{
            value
        }
    })
    value = 'calorie-tracker'
    wrapper.find('select').at(0).simulate('change',{
        target:{
            value
        }
    })
    expect(wrapper.find('option').at(2).prop('value')).toBe('week')
    expect(wrapper.find('option').at(2).text()).toBe('Last Week')
    expect(wrapper.find('option').at(3).prop('value')).toBe('fortnight')
    expect(wrapper.find('option').at(3).text()).toBe('Last 2 Weeks')
    expect(wrapper.find('option').at(4).prop('value')).toBe('2months')
    expect(wrapper.find('option').at(4).text()).toBe('2 Months')
})

test('should call setChartType with default chart and option data',()=>{
    wrapper.find('form').simulate('submit',{
        preventDefault:() => { }
    })
    expect(setChartType).toHaveBeenCalledWith({chart:'calorie-tracker', option:'week'})
})

