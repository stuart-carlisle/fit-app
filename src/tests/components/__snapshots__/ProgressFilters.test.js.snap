// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the progress filters correctly 1`] = `
<div>
  <button
    className="button__back-button"
    onClick={[Function]}
  >
    <img
      className="image__header"
      src="/images/back-arrow.svg"
    />
  </button>
  <div
    className="page-header__filter page-header__filter--with-back"
  >
    <div
      className="content-container"
    >
      <div
        className="header__content--filters"
      >
        <form
          className="filters-text-input__span-container"
          onSubmit={[Function]}
        >
          <div
            className="input-group__item-page-header"
          >
            <select
              className="minimal minimal--chart-filter"
              onChange={[Function]}
              value="calorie-tracker"
            >
              <option
                value="calorie-tracker"
              >
                CALORIE TRACKER
              </option>
              <option
                value="weight-tracker"
              >
                WEIGHT TRACKER
              </option>
            </select>
          </div>
          <div
            className="input-group__item-page-header"
          >
            <select
              className="minimal minimal--chart-filter"
              onChange={[Function]}
              value="week"
            >
              <option
                value="week"
              >
                Last Week
              </option>
              <option
                value="fortnight"
              >
                Last 2 Weeks
              </option>
              <option
                value="2months"
              >
                2 Months
              </option>
            </select>
          </div>
          <button
            className="button filters-button--form-page-header"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;
